src/core/load-fragment.c

```diff
@@ -5423,6 +5423,87 @@ int config_parse_ip_filter_bpf_progs(
         return 0;
 }
 
+int config_parse_restrict_network_interfaces(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        CGroupContext *c = data;
+        bool is_allow_rule = true;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                /* Empty assignment resets the list */
+                c->restrict_network_interfaces = set_free_free(c->restrict_network_interfaces);
+                return 0;
+        }
+
+        if (rvalue[0] == '~') {
+                is_allow_rule = false;
+                rvalue++;
+        }
+
+        if (!c->restrict_network_interfaces) {
+                c->restrict_network_interfaces = set_new(NULL);
+                if (!c->restrict_network_interfaces)
+                        return log_oom();
+
+                /* Only initialize this when creating the set */
+                c->restrict_network_interfaces_is_allow_list = is_allow_rule;
+        }
+
+        for (const char *p = rvalue;;) {
+                _cleanup_free_ char *word = NULL;
+                char *it;
+
+next:
+                r = extract_first_word(&p, &word, NULL, EXTRACT_UNQUOTE);
+                if (r == 0)
+                        break;
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, r,
+                                   "Trailing garbage in %s, ignoring: %s", lvalue, rvalue);
+                        break;
+                }
+
+                if (c->restrict_network_interfaces_is_allow_list != is_allow_rule) {
+                        SET_FOREACH(it, c->restrict_network_interfaces) {
+                                if (streq(it, word)) {
+                                        free(set_remove(c->restrict_network_interfaces, it));
+                                        break;
+                                }
+                        }
```

Comment:
this loop can be replaced by a simple: `set_remove(c->restrict_network_interfaces, word);`